topics/Dialog-and-popup-sizes.md

```diff
@@ -0,0 +1,278 @@
+            </td>
+            <td>
+                <p>Width: min&nbsp;300&nbsp;px</p>
+                <p>Height (when multiple lines): min&nbsp;120&nbsp;px</p>
+            </td>
+        </tr>
+    </table>
+
+3. Select the biggest size out of the two to add the minimal size to a component
+
+#### Example 1
+
+The <control>Rename Inheritors</control> dialog has a table with FQNs. Add 350&nbsp;px as the minimum width for a table column. This will make the dialog wider than the recommended 500&nbsp;px but will show more of FQNs.
+
+<table style="none" border="false">
+    <tr>
+         <td width="706">
+            <format color="369650" style="bold">Correct</format>
+            <img src="dialog_popup_sizes_example_1_correct.png" alt="The Rename Inheritors dialog has a minimum width set for each table column"/>
+         </td>
+    </tr>
+    <tr>
+         <td width="706">
+            <format color="DarkOrange" style="bold">Acceptable</format>
+            <img src="dialog_popup_sizes_example_1_acceptable.png" alt="The Rename Inheritors dialog has the Medium size as default"/>
+         </td>
+    </tr>
+</table>
+
+#### Example 2
+
+<control>Settings</control> dialog has the default size 1000&nbsp;px. In the <control>Code Style settings for Java</control> set 400&nbsp;px as the minimal width of the code editor. This will make the dialog wider but will show more code
+
+<table style="none" border="false">
+    <tr>
+         <td width="706">
+            <format color="369650" style="bold">Correct</format>
+            <img src="dialog_popup_sizes_example_2_correct.png" alt=""/>
+         </td>
+    </tr>
+    <tr>
+         <td width="706">
+            <format color="DarkOrange" style="bold">Acceptable</format>
+            <img src="dialog_popup_sizes_example_2_acceptable.png" alt=""/>
+         </td>
+    </tr>
+</table>
+
+#### Example 3
+
+Add horizontal and vertical 100&nbsp;px insets inside the <control>Diagram</control> popup to make it adaptable to the different amounts of content inside
+
+<table style="none" border="false">
+    <tr>
+         <td width="706">
+            <format color="369650" style="bold">Correct</format>
+            <img src="dialog_popup_sizes_example_3_correct.png" alt=""/>
+         </td>
+    </tr>
+    <tr>
+         <td width="706">
+            <format color="DarkOrange" style="bold">Acceptable</format>
+            <img src="dialog_popup_sizes_example_3_acceptable.png" alt=""/>
+         </td>
+    </tr>
+</table>
+
+## Empty state
+
+Some components, for example, <control>tables</control>, can have an [empty state](empty_state.md) when there is no content. In this case, the minimum size depends on which state takes more space: the component with content or with an empty state.
+
+In most cases, an empty state takes less space than the minimum size of a component. In other cases, the minimum size of the component should be determined by the size of the empty state to reduce unexpected resizing.
```

Comment:
1. I'd suggest adding an example where the component size is defined by its empty state
2. Possibly write explicitly the sizes of the table again here for this particular example — we take min width and height and get this empty state size